corehq/warehouse/tests/test_dimensions.py

```diff
@@ -90,3 +95,85 @@ def test_user_types(self):
             UserDim.objects.filter(user_type=WEB_USER_TYPE).first().user_id,
             'beeboobop',
         )
+
+
+class TestLocationDim(TestCase):
+
+    domain = 'location-dim-test'
+
+    @classmethod
+    def tearDownClass(cls):
+        LocationStagingTable.clear_records()
+        LocationTypeStagingTable.clear_records()
+        LocationDim.clear_records()
+        super(TestLocationDim, cls).tearDownClass()
+
+    def test_location_dim(self):
+        start = datetime.utcnow() - timedelta(days=3)
+        end = datetime.utcnow() + timedelta(days=3)
+        tree = {
+            ('Illinois', 'state'): {
+                ('Naperville', 'city'): {
+                    ('Home', 'home'): {}
+                },
+                ('Chicago', 'city'): {},
+            }
+        }
+        create_location_records_from_tree(self.domain, tree)
+
+        self.assertEqual(LocationStagingTable.objects.count(), 4)
+        self.assertEqual(LocationTypeStagingTable.objects.count(), 3)
+
+        LocationDim.commit(start, end)
+        self.assertEqual(LocationDim.objects.count(), 4)
+        home_location = LocationDim.objects.filter(name='Home').first()
+
+        self.assertEqual(
+            home_location.location_level_2,
+            LocationDim.objects.filter(name='Illinois').first().sql_location_id,
+        )
+        self.assertEqual(
+            home_location.location_level_1,
+            LocationDim.objects.filter(name='Naperville').first().sql_location_id,
+        )
+        self.assertEqual(home_location.location_level_0, home_location.sql_location_id)
```

Comment:
Is there an advantage to having the hierarchy this way round? If the hierarchy was inverted we could just do a query like this:
```
select * from locatoin_dim where location_level_N = <location_id>;
```
It would mean we'd need to know what level the location with id <location_id> is.